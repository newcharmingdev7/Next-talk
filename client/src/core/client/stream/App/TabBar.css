.smallTab {
  padding: var(--spacing-3);
}

.condensedTab {
  padding: var(--spacing-2);
}

.configureTab {
  padding-left: var(--spacing-4);
  padding-right: var(--spacing-4);
}

.smallText {
  font-family: var(--font-family-secondary);
  font-size: var(--font-size-1);
  line-height: 1;

  padding-top: var(--spacing-1);
}
