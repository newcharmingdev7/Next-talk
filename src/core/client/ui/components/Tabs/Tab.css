.root {
  display: inline-block;
  list-style: none;
  margin-right: -1px;
  margin-bottom: -1px;
}

.button {
  height: 100%;
  box-sizing: border-box;
  border-bottom: 0;
  list-style: none;
  padding: var(--spacing-unit);
  font-weight: var(--font-weight-regular);
  font-family: var(--font-family-serif);

  &:hover {
    cursor: pointer;
  }
}

.primary {
  background: var(--palette-grey-lightest);
  color: var(--palette-grey-main);
  border: 1px solid var(--palette-grey-lighter);
  padding: calc(0.5 * var(--spacing-unit)) calc(var(--spacing-unit) * 2);

  &:first-child {
    border-top-left-radius: var(--round-corners);
  }

  &:last-child {
    border-top-right-radius: var(--round-corners);
  }

  &.active {
    background-color: var(--palette-common-white);
    color: var(--palette-common-black);
    border-bottom-color: var(--palette-common-white);
    border-top-width: calc(0.5 * var(--spacing-unit));
    border-top-color: var(--palette-primary-main);
  }
}

.secondary {
  padding: calc(0.5 * var(--spacing-unit)) calc(var(--spacing-unit) * 2);

  &.active {
    font-weight: var(--font-weight-medium);
    border-bottom: 3px solid var(--palette-primary-main);
  }
}
