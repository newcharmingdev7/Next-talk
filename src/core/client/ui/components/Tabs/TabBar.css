.root {
  display: flex;
  padding: 0;
  margin: 0;
}

.primary {
  border-bottom: 1px solid #787d80;
}

.secondary {
  border-bottom: 1px solid #e0e0e0;
}
