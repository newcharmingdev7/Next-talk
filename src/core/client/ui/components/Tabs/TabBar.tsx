import cn from "classnames";
import React, { StatelessComponent } from "react";
import { withStyles } from "talk-ui/hocs";
import * as styles from "./TabBar.css";

export interface TabBarProps {
  /**
   * Convenient prop to override the root styling.
   */
  className?: string;
  /**
   * Override or extend the styles applied to the component.
   */
  classes: typeof styles;

  color: "primary" | "secondary";
  activeId?: string;
  defaultActiveId?: string;
  onChange?: (activeId: string) => void;
  onTabClick?: () => void;
}

const TabBar: StatelessComponent<TabBarProps> = props => {
  const {
    className,
    classes,
    children,
    onTabClick,
    activeId,
    color,
    defaultActiveId,
  } = props;

  const rootClassName = cn(
    classes.root,
    [
      {
        [classes.primary]: color === "primary",
        [classes.secondary]: color === "secondary",
      },
    ],
    className
  );

  const tabs = React.Children.toArray(children).map(
    (child: React.ReactElement<any>, index: number) =>
      React.cloneElement(child, {
        tabId: index,
        active:
          defaultActiveId && !activeId
            ? child.props.tabId === defaultActiveId
            : child.props.tabId === activeId,
        color,
        onTabClick,
      })
  );

  return <ul className={rootClassName}>{tabs}</ul>;
};

TabBar.defaultProps = {
  color: "primary",
};

const enhanced = withStyles(styles)(TabBar);
export default enhanced;
