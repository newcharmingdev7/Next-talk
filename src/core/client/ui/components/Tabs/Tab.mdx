---
name: Tab
menu: UI Kit
---

import { Playground, PropsTable } from 'docz'
import HorizontalGutter from '../HorizontalGutter'


import TabBar from './TabBar'
import Tab from './Tab'

# Tab
`Tab` component is to be used within the `TabBar` component. This renders a Tab inside the Tab Bar.s

## Basic Usage
<Playground>
  <HorizontalGutter>
    <TabBar defaultActiveTab="one">
      <Tab tabId="one">Tab one</Tab>
      <Tab tabId="second">Active Tab</Tab>
      <Tab tabId="three">Tab Three</Tab>
    </TabBar>
  </HorizontalGutter>
</Playground>
