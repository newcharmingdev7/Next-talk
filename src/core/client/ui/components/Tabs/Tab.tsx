import cn from "classnames";
import React, { StatelessComponent } from "react";
import { withStyles } from "talk-ui/hocs";
import * as styles from "./Tab.css";

export interface TabBarProps {
  /**
   * Convenient prop to override the root styling.
   */
  className?: string;
  /**
   * Override or extend the styles applied to the component.
   */
  classes: typeof styles;

  tabId: string;
  active: boolean;
  color: string;
  onTabClick?: () => void;
}

const TabBar: StatelessComponent<TabBarProps> = props => {
  const {
    className,
    classes,
    children,
    tabId,
    active,
    color,
    onTabClick,
  } = props;

  const rootClassName = cn(
    classes.root,
    {
      [classes.primary]: color === "primary",
      [classes.secondary]: color === "secondary",
      [classes.active]: active,
    },
    className
  );

  return (
    <li
      className={rootClassName}
      key={tabId}
      id={`${tabId}-tab`}
      role="tab"
      onClick={onTabClick}
      aria-controls={tabId}
      aria-selected={active}
    >
      {children}
    </li>
  );
};

const enhanced = withStyles(styles)(TabBar);
export default enhanced;
