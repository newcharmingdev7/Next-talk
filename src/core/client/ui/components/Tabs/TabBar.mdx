---
name: TabBar
menu: UI Kit
---

import { Playground, PropsTable } from 'docz'
import HorizontalGutter from '../HorizontalGutter'


import TabBar from './TabBar'
import Tab from './Tab'

# TabBar

## Basic Use
<Playground>
  <HorizontalGutter>

    <TabBar defaultActiveTab="one">
      <Tab tabId="one">Tab one</Tab>
      <Tab tabId="second">Active Tab</Tab>
      <Tab tabId="three">Tab Three</Tab>
    </TabBar>

    <TabBar defaultActiveTab="one" activeTab="second">
      <Tab tabId="one">Tab one</Tab>
      <Tab tabId="second">Active Tab</Tab>
      <Tab tabId="three">Tab Three</Tab>
    </TabBar>

    <TabBar defaultActiveTab="one" activeTab="three">
      <Tab tabId="one">Tab one</Tab>
      <Tab tabId="second">Active Tab</Tab>
      <Tab tabId="three">Tab Three</Tab>
    </TabBar>

  </HorizontalGutter>
</Playground>

## Secondary Color
<Playground>
  <HorizontalGutter>

    <TabBar defaultActiveTab="one" color="secondary">
      <Tab tabId="one">Tab one</Tab>
      <Tab tabId="second">Active Tab</Tab>
      <Tab tabId="three">Tab Three</Tab>
    </TabBar>

  </HorizontalGutter>
</Playground>
