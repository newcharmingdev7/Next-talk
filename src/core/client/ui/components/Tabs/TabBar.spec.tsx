import React from "react";
import TestRenderer from "react-test-renderer";

import Tab from "./Tab";
import TabBar from "./TabBar";

it("renders correctly", () => {
  const renderer = TestRenderer.create(
    <TabBar activeTab="one">
      <Tab tabId="one">One</Tab>
      <Tab tabId="two">Two</Tab>
      <Tab tabId="three">Three</Tab>
    </TabBar>
  );
  expect(renderer.toJSON()).toMatchSnapshot();
});
