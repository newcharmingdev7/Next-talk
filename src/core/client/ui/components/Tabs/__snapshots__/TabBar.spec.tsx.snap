// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="TabBar-root TabBar-primary"
  role="tablist"
>
  <li
    className="Tab-root Tab-primary Tab-active"
    id="one-tab"
    onClick={[Function]}
    role="presentation"
  >
    <button
      aria-controls="one"
      aria-selected={true}
      className="BaseButton-root"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      role="tab"
      type="button"
    >
      One
    </button>
  </li>
  <li
    className="Tab-root Tab-primary"
    id="two-tab"
    onClick={[Function]}
    role="presentation"
  >
    <button
      aria-controls="two"
      aria-selected={false}
      className="BaseButton-root"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      role="tab"
      type="button"
    >
      Two
    </button>
  </li>
  <li
    className="Tab-root Tab-primary"
    id="three-tab"
    onClick={[Function]}
    role="presentation"
  >
    <button
      aria-controls="three"
      aria-selected={false}
      className="BaseButton-root"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      role="tab"
      type="button"
    >
      Three
    </button>
  </li>
</ul>
`;
